### ISSUE TITLE
GhidraProject#getServerRepository(String, int, String, boolean) is ignoring createIfNeeded

### ISSUE THREAD

**Issue by LorenzHetterich** [NationalSecurityAgency/ghidra#3017]
**Describe the bug**
The method [`getServerRepository(String host, int port, String repositoryName, boolean createIfNeeded)`](https://github.com/NationalSecurityAgency/ghidra/blob/3b867b3444afa0d979aab467dcfa901ee6585e14/Ghidra/Features/Base/src/main/java/ghidra/base/project/GhidraProject.java#L153) of class [`ghidra.base.project.GhidraProject`](https://github.com/NationalSecurityAgency/ghidra/blob/3b867b3444afa0d979aab467dcfa901ee6585e14/Ghidra/Features/Base/src/main/java/ghidra/base/project/GhidraProject.java) is ignoring the parameter `createIfNeeded` and always creates the repository if non existent.

**Expected behavior**
If `createIfNeeded` is `false`, I would expect no repository to be created if it doesn't exist.

**Comment by dev747368** [comment]
Considering that the method doesn't seem to be called anywhere, the more appropriate response would probably be to remove the method.

Do you have out-of-tree code that is using this method?

**Comment by LorenzHetterich** [comment]
Yes, We are currently building a Java library providing CTF utilities (it's not public at the moment as most features are not yet implemented).
This library supports working with Ghidra Programs (locally and from ghidra servers) and uses `getServerRepository` internally.
It would not be a huge problem to just copy-paste the code of `getServerRepository` though.

**Comment by dev747368** [comment]
Probably best to do it that way.

**Comment by LorenzHetterich** [comment]
So we'll get rid of the method (at some point in the future)?
